lightning/src/ln/peer_handler.rs

```diff
@@ -389,10 +401,10 @@ impl Peer {
 		self.pending_outbound_buffer.len() < OUTBOUND_BUFFER_LIMIT_READ_PAUSE
 	}
 
-	/// Determines if we should push additional gossip messages onto a peer's outbound buffer for
-	/// backfilling gossip data to the peer. This is checked every time the peer's buffer may have
-	/// been drained.
-	fn should_buffer_gossip_backfill(&self) -> bool {
+	/// Determines if we should push additional messages onto a peer's outbound buffer for backfilling
+	/// onion messages and gossip data to the peer. This is checked every time the peer's buffer may
+	/// have been drained.
+	fn should_buffer_message_backfill(&self) -> bool {
```

Comment:
> Hmmmmm right, now I remember why we were queueing onion messages rather than just draining as required - as currently written we let non-backfill gossip starve onion messages because that is buffered.

I'm confused, currently we treat OMs ~the same as gossip was treated prior to this change. Does this mean gossip backfilling does not currently work as intended in `main`?

Reply:
It looks like we only produce new gossip broadcasts when channels open+close, and on `timer_tick`. So FWIW new gossip broadcasts seem naturally somewhat rate limited, unless the user is specifically spamming the broadcast method

In any case, am I correct that this means we're moving OM sending back to event processing? Because you'd want to only queue->write 1 OM at a time in `do_attempt_write_data` iiuc

Reply:
Of all the methods currently in "Peer", 
this method "should_buffer_message_backfill" is the most deceptive. 
iiuc, it means more like "can_send_init_sync_gossip"  or "should_send_init_sync_gossip" or "should_buffer_init_sync_gossip"

""Determines if we should push additional gossip messages onto a peer's outbound buffer. This is checked every time the peer's buffer may have been drained. 
Note: This is only checked for gossip messages resulting from initial sync. For new gossip broadcasts we either send them immediately on receive or drop them. (acc. to `should_drop_forwarding_gossip`) ""

Please correct it if this understanding is wrong or it can be better renamed. 
Also wanted to remove references to backfill, as it was slightly confusing.

Consider renaming "buffer_full_drop_gossip" => "should_drop_forwarding_gossip", buffer is full is implementation detail

Reply:
@G8XSU Could you move this comment to #1683? I separated out this code change to that PR